### DISCUSSION TITLE
請問群暉安裝DOCKER出錯

### DISCUSSION THREAD

**Discussion by kennychui88** [hslr-s/sun-panel#281] [Q&A]
Docker 出错了: Docker ERROR- [Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?]

**Comment by wintsa123** [comment]
你个人问题。。。我群晖正常

**Comment by xjming1314** [comment]
我来当一个终结者：出现这个问题一般是两种情况，第一个没有配置路径映射。即没有按照作者说明，把/var/run/docker.sock映射到/var/run/docker.sock。  第二个是配置了映射，但是没权限。

1）路径映射有3种办法。 一个是通过群晖套件的社区版资源下载，这里是共享好了。 第二个是通过群晖的docker安装，但是ssh到后台，安装的时候自定义好映射路径。 第三个就是在群晖的docker已经配置好了，不想删除重建，然而可视化更改路径映射不能选择/var/run/docker.sock。

我是第三种。这里就要有点骚操作了。  第一步 ssh后，用这个命令，给docker.sock创建一个符号连接，类似于快捷方式的玩意。 ln -s /var/run/docker.sock /volume2/docker/homepage/。   此时去docker/homepage的后台就能看到docker.sock文件了。 但是群晖的可视化还是无法查看此文件，更不能选中。  

骚操作来了，先在homepag随便新建一个文件叫docker.sock（先把刚刚的符号连接重命名一下，命名为1），然后停止原先的容器，可视化选择文件映射，文件就用刚刚新建的这个docker.sock。  然后保存，不要运行容器。 保存完毕后，删除这个docker.sock，再把原始的docker.sock（被重命名为1的文件）再名字改回来。  改完启动容器即可。

2）配置完映射，还是报错，就是没权限。  那就sudo chmod 666 /var/run/docker.sock  给他个权限。  解决重启失效，就在群晖创建个触发任务，每次开机自动运行这个命令即可
